armi/reactor/blocks.py

```diff
@@ -1519,6 +1535,85 @@ def coords(self, rotationDegreesCCW=0.0):
             round(y, units.FLOAT_DIMENSION_DECIMALS),
         )
 
+    def createCopy(self):
+        """
+        Create a new copy of a block that is less expensive than a full deepcopy.
+
+        Notes
+        -----
+        This can be used to improve performance when a new copy of a reactor needs to be
+        built, but the full detail of the block (including component geometry, material,
+        number density, etc.) is not required for the targeted physics solver being applied
+        to the new reactor model.
+
+        The main use case is for the uniform mesh converter (UMC). Frequently, a deterministic
+        nutronics solver will require a uniform mesh reactor, which is produced by the UMC.
+        Many deterministic solvers for fast spectrum reactors will also treat the individual
+        blocks as homogenized mixtures. Since the neutronics solver does not need to know about
+        the geometric and material details of the individual child components within a block,
+        we can save significant effort while building the uniform mesh reactor with the UMC
+        by omitting this detailed data and only providing the necessary level of detail for
+        the uniform mesh reactor: number densities on each block.
+
+        .. note: Individual components within a block can have different temperatures, and this
+        can affect cross sections. This temperature variation is captured by the lattice physics
+        module. As long as temperature distribution is correctly captured during cross section
+        generation, it doesn't need to be transferred to the neutronics solver directly through
+        this copy operation.
+
+        .. note: If you make a new block, you must add it to an assembly and a reactor.
+
+        See Also
+        --------
+        armi.reactor.converters.uniformMesh.UniformMeshGeometryConverter.makeAssemWithUniformMesh
+        """
+
+        b = self.__class__(self.getName(), height=self.getHeight())
+        b.setType(self.getType())
+
+        # assign macros and LFP
+        b.macros = self.macros
+        b._lumpedFissionProducts = self._lumpedFissionProducts
+        b.p.buGroup = self.p.buGroup
+
+        hexComponent = Hexagon(
+            "homogenizedHex",
+            "Mixture",
+            self.getAverageTempInC(),
+            self.getAverageTempInC(),
+            self._pitchDefiningComponent[1],
+        )
+        emptyNDens = {nuc: 0.0 for nuc in self.getNuclides()}
+        hexComponent.setNumberDensities(emptyNDens)
+        b.add(hexComponent)
```

Comment:
I renamed it to `_createHomogenizedCopy()`.

It's a bit of a misnomer for the base method `Block._createHomogenizedCopy()`, which just performs a `deepcopy` of the block, but that's probably okay. The non-homogenized block will still perform the correct functions, it will just be less efficient than `HexBlock._createHomogenizedCopy()`.